Minor fixes from #7

vladimir-mencl-eresearch opened the pull request:
Hi Zenon,

I have cherry-picked the two commits from #7 that are not related to the InstServer M2M change - please let me know if you are happy to merge these two minor fixes.

Cheers,
Vlad

vladimir-mencl-eresearch committed:
PIP: add PyYAML and Mako for exports and templates

vladimir-mencl-eresearch committed:
Templates: servers_edit: add missing </div>

A missing </div> was at times causing the footer to go out of place and cover
the form contents.

zmousm commented:
Thanks for the template fix.

As for `requirements.txt`:
- Indeed PyYAML is missing; but in order to track the versions of other dependencies (still mostly aligned to Debian Wheezy), I would write the version argument as `>=3.0.10`.
- Mako is only necessary for (example) templates under extras. As that actually won't typically run in the same system/environment as DjNRO, I would be inclined not to add it in `requirements.txt` or perhaps to add it in `optional_requirements.txt`. What do you think?

Also, please rebase this PR on the rewritten master branch.

vladimir-mencl-eresearch committed:
PIP: add requirements-optional.txt

As per discussion in #10:

* Make PyYAML import floating in 3.x (without pinning to 3.0.10)
* Move Mako import to new file requirements-optional.txt
* Update docs/installation/requirements.md to refer to requirements-optional.txt

vladimir-mencl-eresearch commented:
Hi Zenon, 

Most packages are listed as pinned to exact versions - but I've updated the PR as requested and rebased - can you please merge?

Cheers,
Vlad
